artifactory_test.go

```diff
@@ -3122,6 +3122,65 @@ func TestArtifactoryDownloadByBuildUsingSimpleDownloadWithProject(t *testing.T)
 	cleanArtifactoryTest()
 }
 
+func TestArtifactoryDownloadWithEnvProject(t *testing.T) {
+	initArtifactoryProjectTest(t)
+	accessManager, err := utils.CreateAccessServiceManager(serverDetails, false)
+	assert.NoError(t, err)
+	projectKey := "tstprj"
+	// Delete the project if already exists
+	accessManager.DeleteProject(projectKey)
+
+	// Create new project
+	projectParams := accessServices.ProjectParams{
+		ProjectDetails: accessServices.Project{
+			DisplayName: "testProject",
+			ProjectKey:  projectKey,
+		},
+	}
+	err = accessManager.CreateProject(projectParams)
+	assert.NoError(t, err)
+	// Assign the repository to the project
+	err = accessManager.AssignRepoToProject(tests.RtRepo1, projectKey, true)
+	assert.NoError(t, err)
+
+	// Delete the build if exists
+	inttestutils.DeleteBuild(serverDetails.ArtifactoryUrl, tests.RtBuildName1, artHttpDetails)
+
+	specFileB, err := tests.CreateSpec(tests.SplitUploadSpecB)
+	assert.NoError(t, err)
+	buildNumberA := "123"
+	os.Setenv(coreutils.BuildName, tests.RtBuildName1)
```

Comment:
The following goea for all ```os.Setenv``` in this file (not only this function) - let's assert the error received from the this function. The same goes for the ```os.Unsetenv``` calls.